src/bootstrap/src/bin/main.rs

```diff
@@ -108,35 +108,44 @@ fn check_version(config: &Config) -> Option<String> {
         msg.push_str("WARNING: The use of `changelog-seen` is deprecated. Please refer to `change-id` option in `config.example.toml` instead.\n");
     }
 
-    let latest_config_id = CONFIG_CHANGE_HISTORY.last().unwrap();
+    let latest_change_id = CONFIG_CHANGE_HISTORY.last().unwrap().change_id;
+    let warned_id_path = config.out.join("bootstrap").join(".last-warned-change-id");
+
     if let Some(id) = config.change_id {
-        if &id == latest_config_id {
+        if id == latest_change_id {
             return None;
         }
 
-        let change_links: Vec<String> = find_recent_config_change_ids(id)
-            .iter()
-            .map(|id| format!("https://github.com/rust-lang/rust/pull/{id}"))
-            .collect();
-        if !change_links.is_empty() {
-            msg.push_str("WARNING: there have been changes to x.py since you last updated.\n");
-            msg.push_str("To see more detail about these changes, visit the following PRs:\n");
-
-            for link in change_links {
-                msg.push_str(&format!("  - {link}\n"));
+        if let Ok(last_warned_id) = fs::read_to_string(&warned_id_path) {
+            if id.to_string() == last_warned_id {
+                return None;
             }
+        }
+
+        let changes = find_recent_config_change_ids(id);
 
-            msg.push_str("WARNING: there have been changes to x.py since you last updated.\n");
+        if !changes.is_empty() {
+            msg.push_str("There have been changes to x.py since you last updated:\n");
+
+            for change in changes {
+                msg.push_str(&format!("  [{}] {}\n", change.severity.to_string(), change.summary));
+                msg.push_str(&format!(
+                    "    - PR Link https://github.com/rust-lang/rust/pull/{}\n",
+                    change.change_id
+                ));
```

Comment:
This change results this format:

```sh
Building bootstrap
   Compiling bootstrap v0.0.0 (/home/nimda/devspace/onur-ozkan/rust/src/bootstrap)
    Finished dev [unoptimized] target(s) in 4.85s
There have been changes to x.py since you last updated:
  [INFO] New option `rust.parallel-compiler` added to config.toml.
    - PR Link https://github.com/rust-lang/rust/pull/117435
  [INFO] Default value of `download-ci-llvm` was changed for `codegen` profile.
    - PR Link https://github.com/rust-lang/rust/pull/116881
note: to silence this warning, update `config.toml` to use `change-id = 116881` instead
Building stage0 tool tidy (x86_64-unknown-linux-gnu)

```